packages/angular_devkit/build_angular/src/browser/tests/options/inline-critical_spec.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { buildWebpackBrowser } from '../../index';
+import { BASE_OPTIONS, BROWSER_BUILDER_INFO, describeBuilder } from '../setup';
+
+describeBuilder(buildWebpackBrowser, BROWSER_BUILDER_INFO, (harness) => {
+  describe('Option: "inlineCritical"', () => {
+    beforeEach(async () => {
+      await harness.writeFile('src/styles.css', 'body { color: #000 }');
+    });
+
+    it(`should extract critical css when 'inlineCritical' is true`, async () => {
+      harness.useTarget('build', {
+        ...BASE_OPTIONS,
+        optimization: {
+          scripts: false,
+          styles: {
+            minify: true,
+            inlineCritical: true,
+          },
+          fonts: false,
+        },
+        styles: ['src/styles.css'],
+      });
+
+      const { result } = await harness.executeOnce();
+
+      expect(result?.success).toBe(true);
+      harness
+        .expectFile('dist/index.html')
+        .content.toContain(
+          `<link rel="stylesheet" href="styles.css" media="print" onload="this.media='all'">`,
+        );
+      harness.expectFile('dist/index.html').content.toContain(`body{color:#000;}`);
+    });
+
+    it(`should extract critical css when 'inlineCritical' is unset`, async () => {
```

Comment:
```suggestion
    it(`should extract critical css when 'optimization' is unset`, async () => {
```
Should we add  a case for when `inlineCritical` itself is not set but `optimization` is set?

Reply:
added